app/controllers/users_controller.rb

```diff
@@ -163,7 +163,11 @@ def update
   private
 
   def edit_user_params
-    params.require(:user).permit(:email, :first_name, :last_name)
+    params.require(:user).permit(:email, :first_name, :last_name) if @user.admin_editable?
+  end
+
+  def price_group_params
+    params.require(:user).permit(:internal)
```

Comment:
If we're restricting the form element base on user role, we should probably restrict the allowed params as well.